### ISSUE TITLE
[Grid - Sorting] Buttons used to sort are have insufficient names

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3498]
### 😯 Current Behavior

Buttons used for sorting inform only about next sorting method e.g. "Sort ascending". There is no information about what column is being sorted. 

This may be problematic especially if bigger numbers of columns will allow to be sorted.

<img width="739" height="379" alt="Image" src="https://github.com/user-attachments/assets/d442d6de-7b76-419e-a784-1c76269b737e" />

### 🤔 Expected Behavior/possible solution?

Buttons used for sorting should have a name that allows to easily differentiate one from the rest. For this purpose either:
1. the whole `<th>` could be one button that inside have name of column and additional information about next sorting method. Example: https://react.carbondesignsystem.com/?path=/story/components-datatable-sorting--default 
2. name of column should be added to `aria-label` of button so it would look like `aria-label='Student sort ascending'`.

Option 1 is recommended.

### 🖥️ Steps to Reproduce

1. Go to https://storybook.sanomalearning.design/?path=/story/grid-sorting--basic
2. With screen reader turned on check if buttons for sorting inform about column name that will be sorted

### Were you able to reproduce this in our storybook?

Yes

### If yes, provide the storybook URL

https://storybook.sanomalearning.design/?path=/story/grid-sorting--basic

### ⚖️ Impact

Not a huge problem, we have a workaround

### Framework

Other

### What browsers are you seeing the problem on?

All

### Component version

0.7.4

### 👤 Your name

Kacper

### 🧢 Your product/team

SLDS

### 🎨 Your Theme(s)

Sanoma Learning
